kindelia_core/src/node.rs

```diff
@@ -760,9 +760,11 @@ pub fn miner_loop(
 // ----
 
 impl<C: ProtoComm, S: BlockStorage> Node<C, S> {
+  #[allow(clippy::too_many_arguments)]
   pub fn new(
     data_path: PathBuf,
     network_id: u32,
+    addr: C::Address,
```

Comment:
happy to add a comment, but I'm not yet seeing why `addr` should be removed...?

btw, there is presently only one `expect()` in `Node::new()` and it is removed in #250.

When possible, I prefer that `::new()` methods accept all arguments in a *final* state so that calls that can panic are unnecessary within `::new()`.  Just seems cleaner for the API.  So hopefully we can achieve that here.   Sometimes builder pattern is helpful. 

btw, for this PR, the focus is really just on `kindelia_core/net.rs` and `kindelia/main.rs`.  Any changes to other files are just to make those work.  I have other (mostly smaller) PRs coming that focus on eg node.rs, persistence.rs, etc, one at a time.

Reply:
> happy to add a comment (just did), but I'm not yet seeing why addr should be removed...?

Well, I commented this thinking that the `addr: C::Address` parameter will always be related to the `comm: C` parameter, so that whenever you want to use a function that receives both, you would always have to do the same thing and use `comm.get_addr()`.

But in addition, with these parameters separated, the function has the possibility of receiving an address different from the address in `comm`; considering the implementation of `UdpSocket` one could create a struct of this type by binding it to an X address, but passing a Y address to `Node::new`, which would cause confusion in the node, making it having a real address and a nominal address (of course we can simply check if this inconsistency exists the few times the `Node::new` function is called, I'm just saying that there would be that possibility).

Reply:
ok, yeah totally makes sense.  I forgot about comm.get_addr().